src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/ProjectFileSynchronizer.cs

```diff
@@ -43,7 +43,8 @@ public void ProjectFileChanged(string filePath, RazorFileChangeKind kind)
         switch (kind)
         {
             case RazorFileChangeKind.Added:
-                _projectService.AddProject(filePath, rootNamespace: null);
+                // TODO: Do we need to call this at all? And if so, how?!
+                //_projectService.AddProject(filePath);
```

Comment:
On my machine, not having this line has no ill-effects, and logically it seems to me to be completely redundant.

* In the language server, we monitor for the `project.razor.json` file and add the project if it doesn't exist (the .cs file directly above this)
* In OmniSharp we have code in MSBuildProjectManager that adds the project if it doesn't exist
* In VS Mac we have code that hooks up to the DotNetProject instance and adds the project if it doesn't exist
* In VS we have code that hooks up to CPS and adds the project if it doesn't exist
* In the C# extension we don't run this code inside Roslyn, and in rzls see the first dot point

Perhaps there is some nasty edge case that the Razor Developers Of Old™ solved with this project file watching, and we'll now be bitten by it, but I can't think of it. More testing will prove whether this can be removed in the final PRs for this mission.

Reply:
So we only need the remove because file watchers we're handling?

Reply:
I think the same redundancy argument applies to remove, but it just so happens that remove works fine so it was easier to leave, and I didn't want to remove the whole file watching infra, in case it turns out I need to add it back again for some reason.

Long term we either need to:

* care about project file disk changes, and monitor for adds and removes correct
	* I started down this path with an `AddDeferredProject(string projectFileName)` concept, and it would work fine, but requires every "update project" action to provide IntermediateOutputPath if possible, so the deferred project can be hydrated if that happens to be the first update, so didn't want to do it unnecessarily
* don't care about project file disk changes, and remove all of the infrastructure that monitors for it

Reply:
Interestingly, I just ran across https://github.com/dotnet/razor/issues/7942 which was mitigated by supplying the rootNamespace in "as many places as possible". Turns out this spot is the only place where it _couldn't_ be supplied, which makes me think that project file watching was always somewhat questionable. This would also be one of the spots where we end up using the default Razor configuration, instead of the real one, which caused the recent typeparam syntax error issue (though to be fair, there is another spot for that, tracked by https://github.com/dotnet/razor/issues/8905).

I'm leaning more towards removing this entirely and just dealing with the fallout, if any.